### PR TITLE
Allow tito build without .tito directory

### LINKED ISSUE #472: Make `tito build --srpm` work without any .tito/ configuration
To make `tito` the default SCM method in Copr, it would be very convenient
to use it with any random git repository holding just a spec file and
corresponding sources to `git archive`.  I.e. avoid doing the `tito init`
step and `git push`.  Is this, by chance, possible?

### PR DESCRIPTION
**Pull request by FrostyX**
Fix #472

### CODE REVIEW

File: src/tito/cli.py [hunk 1]
```diff
@@ -267,8 +267,24 @@ def main(self, argv):
...
+            "changelog_do_not_remove_cherrypick": 0,
+            "changelog_format": "%s (%ae)",
+        })
+        return config
+
     def load_config(self, package_name, build_dir, tag):
-        self.config = ConfigLoader(package_name, build_dir, tag).load()
+        self.config = self.initial_config
+        if not self.options.without_init:
+            self.config = ConfigLoader(package_name, build_dir, tag).load()
```
Comment (xsuchy) [inline src/tito/cli.py:287]: I think this should be:
```
        if self.options.without_init:
            self.config = self.initial_config
        else:
            self.config = ConfigLoader(package_name, build_dir, tag).load()
```

And the error message of ConfigLoader() should be altered and when it prints:
```
ERROR: Unable to locate branch configuration: /home/msuchy/fedora/git/gnome-logs/rel-eng/tito.props
Please run 'tito init'
```
We should add a line:
```
or run tito --without-init
```

Comment (FrostyX) [reply src/tito/cli.py:289]: Good points, thank you. Updated.

**Review by xsuchy — APPROVED** [review]
+1

**Comment by praiskup** [comment]
Why we can't simply do this even without an explicit `--without-init`?

**Comment by FrostyX** [comment]
> Why we can't simply do this even without an explicit --without-init?

I don't have a strong opinion on this. My reasoning was that running Tito without init, albeit very useful in some cases, it is still a non-typical usage. And I wanted to communicate that to users.

But there is no hard requirement for this, we can make this behavior implicit if we want to.

**Comment by xsuchy** [comment]
> Why we can't simply do this even without an explicit `--without-init`?

Because this way user has to explicitly admit that he is operating with default setting of Tito project. And not with the setting of the git project. 
If it is author, then it encourage him to create configuration (and study what he can do with the configuration).

**Comment by praiskup** [comment]
I don't know. If it is possible, and we can issue a warning to "push the user" toward a useful configuration (while the option could silence the warnings), the explicit option only complicates integrating this feature into higher level tools like Copr (special option needs to be implemented just for this).

**Pull request merged**